### PR TITLE
[stable/fluentd] Session affinity

### PR DESCRIPTION
**Pull request by chrissng**
Signed-off-by: Chris Sng <chrissng@users.noreply.github.com>

#### Is this a new chart
No
> NOTE: We're experiencing a high volume of PRs to this repo and reviews will be delayed. Please host your own chart repository and submit your repository to the Helm Hub instead of this repo to make them discoverable to the community. [Here](https://github.com/helm/hub/blob/master/Repositories.md) is how to submit new chart repositories to the Helm Hub.

#### What this PR does / why we need it:
Allow configuring session affinity on the Service resource. This will ensure that connections from a particular client are forwarded to the same pod. Can be useful in avoiding filename clashes when multiple pods output logs to a common store like S3.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by chrissng** [comment]
@rendhalver @hectorj2f Appreciate an ok-to-test and a quick review please.

File: stable/fluentd/templates/service.yaml [hunk 1]
```diff
@@ -30,6 +30,12 @@ spec:
       targetPort: metrics
       protocol: TCP
   {{- end }}
+  {{- if eq .Values.service.sessionAffinity "ClientIP" }}
+  sessionAffinity: {{ .Values.service.sessionAffinity }}
```
Comment (hectorj2f) [inline stable/fluentd/templates/service.yaml:33]: I would have instead a `None` as default value in values.yaml for a property `sessionAffinity` and include the if `ClientIP` below for sessionAffinityConfig.

**Review by hectorj2f — CHANGES_REQUESTED** [review]

File: stable/fluentd/templates/service.yaml (hunk 1 above)
Comment (hectorj2f) [reply stable/fluentd/templates/service.yaml:33]: That is how it is done in other charts.

Comment (chrissng) [reply stable/fluentd/templates/service.yaml:33]: Thanks for the tip, I've fixed the if condition. `sessionAffinity`  defaults to `None` in `values.yaml` so we are good here.

**Review by hectorj2f — COMMENTED** [review]
/ok-to-test

**Comment by chrissng** [comment]
Hi @hectorj2f, if this PR is good to go, appreciate to have it approved and merged. 🙏

**Comment by chrissng** [comment]
@hectorj2f Can we have this PR merged please?

**Pull request closed without merging**